### PR TITLE
Tidy up code

### PR DESCRIPTION
**Pull request by t3chguy**
Subset of https://github.com/vector-im/element-builder/pull/72 with install/update path confusion fixed

### CODE REVIEW

File: src/artifacts.ts [hunk 1]
```diff
@@ -45,10 +46,51 @@ export function copyAndLog(src: string, dest: string, logger: Logger): Promise<v
...
+
+export async function updateSymlink(target: string, symlink: string, logger: Logger): Promise<void> {
+    logger.info(`Update latest symlink ${symlink} -> ${target}`);
+    try {
+        await fsProm.unlink(symlink);
+    } catch (e) {
+        // probably just didn't exist
+        logger.info("Failed to remove latest symlink", e);
+    }
+    await fsProm.symlink(target, symlink, 'file');
```
Comment (SimonBrandner) [inline src/artifacts.ts:95]: /me is confused - will this work after we have `unlink`ed it?

File: src/desktop_develop.ts [hunk 2]
```diff
@@ -294,25 +294,16 @@ export default class DesktopDevelopBuilder {
...
-                );
-            }
+            const distPath = path.join(repoDir, 'dist');
+            const targetInstallPath = path.join(this.appPubDir, 'install', 'macos');
+            const targetUpdatePath = path.join(this.appPubDir, 'update', 'macos');
+
+            await fsProm.mkdir(targetInstallPath, { recursive: true });
+            await fsProm.mkdir(targetUpdatePath, { recursive: true });
+
+            // Be consistent with windows and don't bother putting the version number in the installer
```
Comment (SimonBrandner) [inline src/desktop_develop.ts:304]: Arguably, we should add it everywhere?

**Review by SimonBrandner — COMMENTED** [review]
Looking great

File: src/desktop_develop.ts (hunk 2 above)
Comment (t3chguy) [reply src/desktop_develop.ts:304]: This PR is merely refactoring, also this is develop so growing a million installers is not great

File: src/artifacts.ts (hunk 1 above)
Comment (t3chguy) [reply src/artifacts.ts:95]: This is moving the existing code, it removes an existing link and creates a new one, there's no way to update/overwrite a symlink without first deleting it from within Node

Comment (SimonBrandner) [reply src/artifacts.ts:95]: Ah! /me was confused about what each argument meant...

**Review by SimonBrandner — APPROVED** [review]

**Pull request merged**